File: naddu/naddu_2/c/killers.c
#include "killers.h"

void killers_clear(void) {
  for (int i = 0; i < MAX_PLY; i++) {
    nodes[i].killer = 0;
  }
}

void killer_set(Node *node, uint32_t move) {
  if (move & MOVE_FLAG_CAPTURE)
    return;
  node->killer = move;
}

File: naddu/naddu_2/src/killers.js

function killersClear() {
  for (let i = 0; i < MAX_PLY; i++) {
    nodes[i].killer = 0;
  }
}

function killerSet(node, move) {
  if (move & MOVE_FLAG_CAPTURE)
    return;
  node.killer = move;
}

